yt_dlp/extractor/youtube.py

```diff
@@ -644,6 +644,28 @@ def _report_alerts(self, alerts, expected=True):
     def _extract_and_report_alerts(self, data, *args, **kwargs):
         return self._report_alerts(self._extract_alerts(data), *args, **kwargs)
 
+    def _extract_badges(self, renderer: dict):
+        badges = set()
+        for badge in try_get(renderer, lambda x: x['badges'], list) or []:
+            label = try_get(badge, lambda x: x['metadataBadgeRenderer']['label'], compat_str)
+            if label:
+                badges.add(label.lower())
+        return badges
+
+    @staticmethod
+    def _join_text_entries(runs):
```

Comment:
This is a duplicate of the `get_text` function

Reply:
That's true. I think there a lot of duplicates of similar code over youtube.py, functions or not. I'll prob look into writing one general function to handle them all.

Reply:
I think I'll deal with this in #505 after this PR is merged